pkg/ucp/frontend/controller/resourcegroups/util.go

```diff
@@ -59,38 +61,161 @@ func (e *InvalidError) Is(err error) bool {
 	return ok
 }
 
-// ValidateDownstream can be used to find and validate the downstream URL for a resource.
-// Returns NotFoundError for the case where the plane or resource group does not exist.
-// Returns InvalidError for cases where the data is invalid, like when the resource provider is not configured.
-func ValidateDownstream(ctx context.Context, client store.StorageClient, id resources.ID) (*url.URL, error) {
+// ValidateRadiusPlane validates that the plane specified in the id exists. Returns NotFoundError if the plane does not exist.
+func ValidateRadiusPlane(ctx context.Context, client store.StorageClient, id resources.ID) (*datamodel.RadiusPlane, error) {
 	planeID, err := resources.ParseScope(id.PlaneScope())
 	if err != nil {
 		// Not expected to happen.
 		return nil, err
 	}
+
 	plane, err := store.GetResource[datamodel.RadiusPlane](ctx, client, planeID.String())
 	if errors.Is(err, &store.ErrNotFound{}) {
 		return nil, &NotFoundError{Message: fmt.Sprintf("plane %q not found", planeID.String())}
 	} else if err != nil {
 		return nil, fmt.Errorf("failed to find plane %q: %w", planeID.String(), err)
 	}
 
+	return plane, nil
+}
+
+// ValidateResourceGroup validates that the resource group specified in the id exists (if applicable).
+// Returns NotFoundError if the resource group does not exist.
```

Comment:
Why would a resource ID not have a resource group? Would it not be mandatory, since as part of radius application it would have a resource group in its id?

Reply:
There are lots of cases like this in the UCP code. A simple one is LIST operations at plane-scope. 

`/planes/radius/local/providers/Applications.Core/containers` => list all containers in my plane/tenant.

This is a valid resource id, but it doesn't have a resource group.